src/goto-programs/show_goto_functions_json.cpp

```diff
@@ -46,12 +46,12 @@ Function: show_goto_functions_jsont::show_goto_functions
 
  Outputs:
 
- Purpose: Walks through all of the functions in the program and outputs
-          JSON
+ Purpose: Walks through all of the functions in the program and returns
+          a JSON object representing all their functions
 
 \*******************************************************************/
 
-void show_goto_functions_jsont::show_goto_functions(
+json_objectt show_goto_functions_jsont::get_goto_functions(
```

Comment:
It seems `jsont` will need to have C++11 rvalue references to make returning such objects by value cheap.